satpy/readers/file_handlers.py

```diff
@@ -122,15 +122,114 @@ def sensor_names(self):
+            return True
+        elif self.filetype_info['file_type'] in ds_ftype:
+            return True
+        return None
+
+    def available_datasets(self, configured_datasets=None):
+        """Get information of available datasets in this file.
+
+        This is used for dynamically specifying what datasets are available
+        from a file in addition to what's configured in a YAML configuration
+        file. Note that this method is called for each file handler for each
+        file type; care should be taken when possible to reduce the amount
+        of redundant datasets produced.
+
+        This method should **not** update values of the dataset information
+        dictionary **unless* this file handler has a matching file type
+        (the data could be loaded from this object in the future) and at least
+        **one** :class:`satpy.dataset.DatasetID` key is also modified.
+        Otherwise, this file type may override the information provided by
+        a more preferred file type (as specified in the YAML file).
+        It is recommended that any non-ID metadata be updated during the
+        :meth:`BaseFileHandler.get_dataset` part of loading.
+        This method is not guaranteed that it will be called before any
+        other file type's handler.
+        The availability "boolean" not being ``None`` does not mean that a
+        file handler called later can't provide an additional dataset, but
+        it must provide more identifying (DatasetID) information to do so
+        and should yield its new dataset in addition to the previous one.
+
+        Args:
+            configured_datasets (list): Series of (bool or None, dict) in the
+                same way as is returned by this method (see below). The bool
+                is whether or not the dataset is available from at least one
+                of the current file handlers. It can also be ``None`` if
+                no file handler knows before us knows how to handle it.
+                The dictionary is existing dataset metadata. The dictionaries
+                are typically provided from a YAML configuration file and may
+                be modified, updated, or used as a "template" for additional
+                available datasets. This argument could be the result of a
+                previous file handler's implementation of this method.
+
+        Returns: Iterator of (bool or None, dict) pairs where dict is the
+            dataset's metadata. If the dataset is available in the current
+            file type then the boolean value should be ``True``, ``False``
+            if we **know** about the dataset but it is unavailable, or
+            ``None`` if this file object is not responsible for it.
+
+        Example 1 - Supplement existing configured information::
+
+            def available_datasets(self, configured_datasets=None):
+                "Add information to configured datasets."
```

Comment:
I guess this should be `"""Add information..."""`

Reply:
I can't do triple quotes inside the actual docstring. This was the best compromise I could come up with (instead of escaping quotes).

Reply:
Yes of course, my bad